### ISSUE TITLE
Update samba version (currently v4.12.2)

### ISSUE THREAD

**Issue by 1moere1** [dperson/samba#393]
The version is pretty old (1.5 years). Please rebuild the image and release a new version.

**Comment by BrodyStone21** [comment]
The image hasn't been updated in 2 months, does @dperson still maintain it or is it just pretty stable at the moment and he hasn't seen a need for changes?

**Comment by crazy-max** [comment]
An up-to-date multi-platform image is available here if you're interested: https://github.com/crazy-max/docker-samba

**Comment by BrodyStone21** [comment]
@crazy-max is this a fork of dperson/samba? What are the advantages of using that over this?

**Comment by Erriez** [comment]
> The image hasn't been updated in 2 months, does @dperson still maintain it or is it just pretty stable at the moment and he hasn't seen a need for changes?

@BrodyStone21  Security updates are required as the images on DockerHub are pushed around March 2021:
https://hub.docker.com/r/dperson/samba/tags

I've created a PR #397, but @dperson needs to rebuild and push new images to DockerHub.

I'm not sure if this project is still maintained and noticed issue #375 `Is this project dead?`

@crazy-max I miss functionality to announce the Samba server on Windows and Linux networks, issue #398.

For this reason, I've created a fork, updated base images, documentation and created new multi-architecture images with `buildx` in my fork: https://github.com/Erriez/docker-samba

My differences:
- Updated images with `alpine:3.15` to fix critical vulnerabilities.
- Update `docker-compose.yml` according updated Docker formatting.
- Add Windows Network Discovery example with WSDD
- Add Linux Network Discovery example with Avahi
- Add multiarchitecture `buildx` support.
- AMD64, ARM64, ARMv6 and ARMv7 images on [DockerHub](https://hub.docker.com/r/erriez/samba)

**Comment by crazy-max** [comment]
@Erriez 

> I miss functionality to announce the Samba server on Windows and Linux networks, issue #398.

Yes this should be part of a sidecar container, see https://github.com/crazy-max/docker-samba/issues/1

@BrodyStone21

Not a fork, from scratch. See https://github.com/crazy-max/docker-samba#features

**Comment by Erriez** [comment]
Looks like a great and useful project, @crazy-max. I'll check if I can use it as alternative of my fork. Maybe I can contribute to crazy-max/docker-samba#1 as I already have this up and running. Then I can drop my fork. Stay tuned.

**Comment by crazy-max** [comment]
> Maybe I can contribute to [crazy-max/docker-samba#1](https://github.com/crazy-max/docker-samba/issues/1) as I already have this up and running. Then I can drop my fork. Stay tuned.

Sure but I wonder if the avahi sidecar container should not be part of a dedicated repository. Then we could add it in a new example like https://github.com/crazy-max/docker-samba/tree/master/examples/compose. Let's discuss about it in https://github.com/crazy-max/docker-samba/issues/1.

**Comment by alpe12** [comment]
@dperson Are you ok?

**Comment by chief8192** [comment]
It appears that the current version (v4.12.2) has a remote code execution vulnerability:
- https://www.samba.org/samba/security/CVE-2021-44142.html 
- https://thehackernews.com/2022/01/new-samba-bug-allows-remote-attackers.html
- https://portswigger.net/daily-swig/critical-samba-flaw-presents-code-execution-threat

**Comment by Silberling** [comment]
It feels like this repo is dead but has a pretty high number of users. Recent patches are crucial for security.

I'd advice to update the container internals manually at least for now:

	sudo docker exec ... - ash
	apk update
	apk upgrade
	exit

then restart the container. Apply again after each deploy / rebuild.

**Comment by unixfox** [comment]
Hi, I just published an up-to-date image here: https://quay.io/repository/unixfox/samba
It is updated at every new version of samba.

**Comment by davidandreoletti** [comment]
> Hi, I just published an up-to-date image here: https://quay.io/repository/unixfox/samba
> It is updated at every new version of samba.

@unixfox Do you have git repo with the update Dockerfile update (if any) ?

**Comment by unixfox** [comment]
> > Hi, I just published an up-to-date image here: [quay.io/repository/unixfox/samba](https://quay.io/repository/unixfox/samba)
> > It is updated at every new version of samba.
> 
> @unixfox Do you have git repo with the update Dockerfile update (if any) ?

I use the original Dockerfile but the github actions workflow file is here: https://github.com/unixfox/periodic-build-with-github-actions/blob/master/.github/workflows/docker-build-samba.yml

**Comment by michaelkrieger** [comment]
Just moved to https://github.com/ServerContainers/samba which seems to have a lot more configuration options, includes wsdd2 daemon built in, and exports AVHAI announcements to the host system's daemon (which most run... or it has its own too).

Would recommend and an easy config change.
